### DISCUSSION TITLE
Deprecate in favor of "Weld JUnit 5 (Jupiter) Extensions"

### DISCUSSION THREAD

**Discussion by SchulteDev** [cdi-unit/cdi-unit#303] [Polls]
Hello, I came across this project by chance. 
cdi-unit seems to be a great project that certainly had its reason for being. But I think that time is over. Weld provides its own Junit Jupiter extensions, so cdi-unit is obsolete.
https://github.com/weld/weld-testing/blob/master/junit5/README.md

**Comment by SchulteDev** [comment]
I hope nobody feels uncomfortable with my suggestion. Open source thrives on projects like these that contribute to the community.

**Comment by fwelland** [comment]
TL;DR:  I am neutral and think maybe both should continue.   

Been using CDI-UNIT (jglue variant)  for some time (with JUNIT4).    Recently started with newer CDI-UNIT for Jakarta namespace 
reasons.    

NOTE:   my/our applicatione of UT (unit testing),  IT (integration testing) **_is well over 90% Spock_**.      IT is only space where CDI-UNIT (or something like that) has _maybe_ some role to play in the verification process (yes, sure debate that all you want ...). 

'Old' CDI-UNIT and this reborn one, don't really play with Spock at all  -- so we don't do DI in most of our ITs and used Spock & Groovy along with some mocking and 'manual' inject  (via property initialization or setters) ;  and, well,  none in UTs.     We did have some token JUNIT4 style tests that used CDI-UNIT to spot check a few things.  

I was pretty interested in the weld-testing for (at least) 2 reasons:   firstly  it has a Spock extension that as documented looks pretty solid  & it has JUNIT Jupiter support  (in our small remaining bundle of JUNIT was at least trying to move to Jupiter). 

However,  **_in my case_**,  it is not easy to take a JUNIT (with CDI-UNIT) and refactor it to do use weld-testing  (or for that matter ditch the JUNIT in favor of a Spock spec 'test-a-like' with weld-testing for Spock).     

This maybe because the subject under test is a BDU/JAR -- and maybe that requires more setup or configuration.   However, it was plagued with un-resolved deps issues.     While CDI-UNIT didn't really have any problem with that.       

Another point:  this experience was via builds using **_gradle 8.8 & 8.9 and test-suites_** -- test-suites go to great lengths to isolate the subject (in my case a  bdu/jar) from the main build.    Maybe weld-testing needed some more configuration that I failed to try/do, while CDI-UNIT just worked. 

YMMV! 

(As time permits, I hope to go back to weld-testing and try to get its Spock extension to work in my environment -- that alone would 'tip the scales for me'.      One could read that as a subtle feature request ;)   ) .

**Comment by ikysil** [comment]
Hi @fwelland 

TY for an elaborate post.
Found your old post at StackOverflow on the subject of [integrating CDI and Spock](https://stackoverflow.com/questions/29749847/how-to-do-cdi-se-from-spock-test).

What particular integration you may expect between Specification instance and CDI container beside injecting the dependencies?

**Reply by fwelland** [reply]
Well that is fair question.    All of CDI, of course!    But more realistically  - injection is the big one and with producers and/or naming qualifiers.      Alternatives are important too -- as in IT world we sub in alternative producers for some stuff like DB resources and such.      Also we do a small amount of interceptors -- albeit I test them in isolation at the moment -- I don't really explicitly test to see that a container actually fires them.   

From that post years gone by -- I guess what I was going for, mainly,  was easy to use way to satisfy injects and still have all/most of the BDD like 'feel' and simplicity of Spock Specs.       The final working thing I made, was a bit close.  NOTE:  I never really applied this anywhere outside of the POC that spawned that SO thread.     The Weld-Testing spock extension seems to present a pretty good pattern.

HTH!

**Comment by ikysil** [comment]
Proof of concept for Spock support - https://github.com/cdi-unit/cdi-unit/pull/327

**Reply by ikysil** [reply]
@fwelland 

POC has been merged - see #331 .

**Reply by fwelland** [reply]
I've been watching!    I hope to kick tires on this in the next few days.     Thanks!
